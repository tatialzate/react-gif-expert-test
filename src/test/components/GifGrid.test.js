import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe( 'Pruebas componente GifGrid', () => {

    test( 'Debe mostrarse el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GifGrid category="dogs" />);
        expect( wrapper ).toMatchSnapshot();
    })
    
    test( 'Debe presentar las imÃ¡genes', () => {
        const gifs = [{
            id: 'ABC',
            title: 'Mi primer Gif',
            url: 'https://www.google.com/image1'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow( <GifGrid category="dogs" />);
        expect(wrapper.find('p').exists()).toBe( false );
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length );
    })
})