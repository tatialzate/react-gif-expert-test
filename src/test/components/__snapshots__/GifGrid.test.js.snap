// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente GifGrid Debe mostrarse el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    dogs
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
